packages/repository-tests/src/crud/relations/integration/belongs-to.factory.integration.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository-tests
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  BelongsToAccessor,
+  BelongsToDefinition,
+  createBelongsToAccessor,
+  Entity,
+  EntityCrudRepository,
+  EntityNotFoundError,
+  Getter,
+  juggler,
+  ModelDefinition,
+  RelationType,
+} from '@loopback/repository';
+import {expect} from '@loopback/testlab';
+import {
+  CrudFeatures,
+  CrudRepositoryCtor,
+  CrudTestContext,
+  DataSourceOptions,
+} from '../../..';
+import {
+  deleteAllModelsInDefaultDataSource,
+  withCrudCtx,
+} from '../../../helpers.repository-tests';
+
+export function belongsToFactorySuite(
+  dataSourceOptions: DataSourceOptions,
+  repositoryClass: CrudRepositoryCtor,
+  features: CrudFeatures,
+) {
+  // Given a Customer and Order models - see definitions at the bottom
+  let db: juggler.DataSource;
+  let customerRepo: EntityCrudRepository<
+    Customer,
+    typeof Customer.prototype.id
+  >;
+  let orderRepo: EntityCrudRepository<Order, typeof Order.prototype.id>;
+
+  describe('BelongsTo relation (integration)', () => {
+    let findCustomerOfOrder: BelongsToAccessor<
+      Customer,
+      typeof Order.prototype.id
+    >;
+
+    before(deleteAllModelsInDefaultDataSource);
+
+    before(
+      withCrudCtx(async function setupRepository(ctx: CrudTestContext) {
+        db = ctx.dataSource;
+        givenCrudRepositories();
```

Comment:
As I commented earlier, this needs to accept `repositoryClass` so that the repositories are inheriting from the right Repository implementation.